src/util/ecdsa.rs

```diff
@@ -413,6 +417,66 @@ impl<'de> ::serde::Deserialize<'de> for PublicKey {
     }
 }
 
+/// An ECDSA signature with the corresponding hash type.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct EcdsaSig {
+    /// The underlying ECDSA Signature
+    pub sig: secp256k1::Signature,
+    /// The corresponding hash type
+    pub hash_ty: EcdsaSigHashType,
+}
+
+impl EcdsaSig {
+
+    /// Deserialize from slice
+    pub fn from_slice(sl: &[u8]) -> Result<Self, EcdsaSigError> {
+        match sl.split_last() {
```

Comment:
`let (hash_ty, sig) = sl.split_last().ok_or(EcdsaSigError::InvalidEcdsaSig)?;`
and get rid of indentation